src/main/kotlin/apply/ui/admin/mail/MailsFormView.kt

```diff
@@ -0,0 +1,81 @@
+package apply.ui.admin.mail
+
+import apply.application.EvaluationService
+import apply.application.MailHistoryService
+import apply.application.MailTargetService
+import apply.application.RecruitmentService
+import apply.application.UserService
+import apply.ui.admin.BaseLayout
+import com.vaadin.flow.component.Component
+import com.vaadin.flow.component.UI
+import com.vaadin.flow.component.button.Button
+import com.vaadin.flow.component.orderedlayout.FlexComponent
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout
+import com.vaadin.flow.component.orderedlayout.VerticalLayout
+import com.vaadin.flow.router.BeforeEvent
+import com.vaadin.flow.router.HasUrlParameter
+import com.vaadin.flow.router.Route
+import com.vaadin.flow.router.WildcardParameter
+import org.springframework.boot.autoconfigure.mail.MailProperties
+import support.views.EDIT_VALUE
+import support.views.FORM_URL_PATTERN
+import support.views.Title
+import support.views.createContrastButton
+import support.views.createPrimaryButton
+
+@Route(value = "admin/mails", layout = BaseLayout::class)
+class MailsFormView(
+    userService: UserService,
+    recruitmentService: RecruitmentService,
+    evaluationService: EvaluationService,
+    mailTargetService: MailTargetService,
+    private val mailHistoryService: MailHistoryService,
+    mailProperties: MailProperties
+) : VerticalLayout(), HasUrlParameter<String> {
+    private val mailForm: MailForm = MailForm(
+        userService,
+        recruitmentService,
+        evaluationService,
+        mailTargetService,
+        mailProperties
+    )
+    private val submitButton: Component = createSubmitButton()
+
+    init {
+        add(Title("메일"), mailForm, createButtons())
+    }
+
+    override fun setParameter(event: BeforeEvent, @WildcardParameter parameter: String) {
+        val result = FORM_URL_PATTERN.find(parameter)
+        result?.let {
+            val (id, value) = it.destructured
+            if (value == EDIT_VALUE) {
+                mailForm.fill(mailHistoryService.getById(id.toLong()))
+                this.submitButton.isVisible = false
+            }
+        } ?: UI.getCurrent().page.history.back() // TODO: 에러 화면을 구현한다.
+    }
```

Comment:
```suggestion
    override fun setParameter(event: BeforeEvent, @WildcardParameter parameter: String) {
        val result = FORM_URL_PATTERN.find(parameter) ?: return goBack()
        val (id, value) = result.destructured
        if (value == EDIT_VALUE) {
            mailForm.fill(mailHistoryService.getById(id.toLong()))
            this.submitButton.isVisible = false
        }
    }

    private fun goBack() {
        UI.getCurrent().page.history.back() // TODO: 에러 화면을 구현한다.
    }
```

depth가 적을수록 보기 좋다고 생각합니다 ㅎㅎ